src/Castle.Windsor/MicroKernel/DefaultKernel.cs

```diff
@@ -278,9 +279,22 @@ IHandler IKernelInternal.CreateHandler(ComponentModel model)
 
 		public virtual IKernel AddFacility(IFacility facility)
 		{
-#pragma warning disable 612,618
-			return AddFacility(facility != null ? facility.GetType().FullName : null, facility);
-#pragma warning restore 612,618
+			if (facility == null)
+			{
+				throw new ArgumentNullException("facility");
+			}
+			var facilityType = facility.GetType();
+			if (facilities.Any(f => f.GetType() == facilityType))
+			{
+				throw new ArgumentException(
+					string.Format(
+						"Facility of type '{0}' has already been registered with the container. Only one facility of a given type can exist in the container.",
+						facilityType.FullName));
+			}
+			facilities.Add(facility);
+			facility.Init(this);
+
+			return this;
 		}
```

Comment:
After fixing this method to deprecate [the old one](https://github.com/castleproject/Windsor/pull/377/commits/161818cd63c2d33cd59d4090467f90d7a45d7663#diff-e1f78c645f8ca7366eb6563394edda3fL25), I came to the conclusion that IConfiguration as a parameter for the IFacility.Init method was no longer needed.

Reply:
Might have been a mistake. Not sure.